### PR TITLE
feat(alert): add an option to use a reverse proxied Telegram API server

### PR DESCRIPTION
**Pull request by RuioWolf**
# Description

TLDR: I add an option for user to use official Telegram API or their own Telegram API by hosting a reverse proxy server. Please give it a test by install a new server, I only tested on my existing tf2server.

One year ago I submit #2805 about using a proxy string for `curlcustomstring`. But the fact is in some area, most IDC will check whether your server is running proxy software to bypass the firewall. While running a socks5 proxy will be extremely easy to be spotted, it's much more reliable to run a reverse proxied Telegram API server somewhere else.

## Type of change

* [ ] Bug fix (a change which fixes an issue).
* [x] New feature (change which adds functionality).
* [ ] New Server (new server added).
* [ ] Refactor (restructures existing code).
* [ ] Comment update (typo, spelling, explanation, examples, etc).

## Checklist

PR will not be merged until all steps are complete.

* [ ] This pull request links to an issue.
* [x] This pull request uses the `develop` branch as its base.
* [x] This pull request Subject follows the Conventional Commits standard.
* [x] This code follows the style guidelines of this project.
* [x] I have performed a self-review of my code.
* [x] I have checked that this code is commented where required.
* [x] I have provided a detailed with enough description of this PR.
* [x] I have checked If documentation needs updating.

## Documentation

I don't know if the document for config a reverse proxy is needed.

### CODE REVIEW

**Review by h3o66 — APPROVED** [review]
Looks good to me

**Comment by RuioWolf** [comment]
Perfect!

**Review by dgibbs64 — APPROVED** [review]
Happy with this 

@RuioWolf  please could you raise an issue for this and link it to this PR. Thank you
